app/client/keybase/debug/keystore.go

```diff
@@ -73,28 +66,55 @@ func initializeDebugKeybase() error {
 
 	// Add validator addresses if not present
 	if len(curAddr) < numValidators {
-		fmt.Println("Rehydrating keybase from private-keys.yaml ...")
+		logger.Global.Debug().Msgf(fmt.Sprintf("Debug keybase initializing... Adding %d validator keys to the keybase", numValidators-len(curAddr)))
+
 		// Use writebatch to speed up bulk insert
 		wb := db.NewWriteBatch()
+
+		// Create a channel to receive errors from goroutines
+		errCh := make(chan error, numValidators)
+
+		// Create a WaitGroup to wait for all goroutines to finish
+		var wg sync.WaitGroup
+		wg.Add(numValidators)
+
 		for _, privHexString := range validatorKeysPairMap {
-			// Import the keys into the keybase with no passphrase or hint as these are for debug purposes
-			keyPair, err := cryptoPocket.CreateNewKeyFromString(privHexString, "", "")
-			if err != nil {
-				return err
-			}
-
-			// Use key address as key in DB
-			addrKey := keyPair.GetAddressBytes()
-
-			// Encode KeyPair into []byte for value
-			keypairBz, err := keyPair.Marshal()
-			if err != nil {
-				return err
-			}
-			if err := wb.Set(addrKey, keypairBz); err != nil {
-				return err
-			}
+			go func(privHexString string) {
+				defer wg.Done()
+
+				// Import the keys into the keybase with no passphrase or hint as these are for debug purposes
+				keyPair, err := cryptoPocket.CreateNewKeyFromString(privHexString, "", "")
+				if err != nil {
+					errCh <- err
+					return
+				}
+
+				// Use key address as key in DB
+				addrKey := keyPair.GetAddressBytes()
+
+				// Encode KeyPair into []byte for value
+				keypairBz, err := keyPair.Marshal()
+				if err != nil {
+					errCh <- err
+					return
+				}
+				if err := wb.Set(addrKey, keypairBz); err != nil {
+					errCh <- err
+					return
+				}
+			}(privHexString)
 		}
+
+		// Wait for all goroutines to finish
+		wg.Wait()
+
+		// Check if any goroutines returned an error
+		select {
+		case err := <-errCh:
```

Comment:
For my personal understanding, did you go with an errChannel instead of a basic list to avoid concurrent writes?

Reply:
Exactly. Without a channel it would have needed a mutex or similar to synchronize access.